cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -176,17 +178,47 @@ private void ensureJDBCDriverIsAvailable() throws Exception {
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
                 dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(driverClass.newInstance());
+      }
       DBUtils.deregisterAllDrivers(driverClass);
       DriverManager.registerDriver(driverShim);
     }
   }
 
+  private boolean tableNameExists() {
+    try {
+      try {
+        ensureJDBCDriverIsAvailable();
+      } catch (Exception e) {
+        Preconditions.checkArgument(false, "Driver must be able to connect");
```

Comment:
Why not just propagate the exception? You can use `throw Throwables.propagate(e)`.